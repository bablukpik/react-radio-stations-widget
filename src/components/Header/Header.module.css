.widgetHeader {
  height: 160px;
  width: 100%;
  background-color: #edae61;
  border-top-left-radius: 47px;
  border-top-right-radius: 47px;
  display: flex;
  align-items: center;
}

.nav {
  width: 100%;
  margin-left: 50px;
  margin-right: 30px;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.navHeading {
  font-size: 44px;
  font-weight: 400;
}
