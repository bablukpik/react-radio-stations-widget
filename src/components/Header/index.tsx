import * as React from 'react'
import Image from '../Image'
import styles from './Header.module.css'
import BackArrow from '../../assets/icons/back-arrow.png'
import Switch from '../../assets/icons/switch.png'
import Button from '../Button'

const Header: React.FC = () => {
  return (
    <header className={styles.widgetHeader}>
      <nav className={styles.nav}>
        <Button>
          <Image
            data-testid="backArrowIcon"
            src={BackArrow}
            alt="Back arrow icon"
          />
        </Button>
        <h1 data-testid="navHeading" className={styles.navHeading}>
          STATIONS
        </h1>
        <Button>
          <Image
            data-testid="switchIcon"
            src={Switch}
            alt="Switch icon"
          />
        </Button>
      </nav>
    </header>
  )
}

export default Header
