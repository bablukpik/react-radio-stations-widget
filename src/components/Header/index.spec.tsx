import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Header from './index';

describe('Header', () => {
  it('should render', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders back arrow icon', () => {
    const { getByTestId, getByAltText} = render(<Header />);
    expect(getByTestId('backArrowIcon')).toBeInTheDocument();
    expect(getByAltText('Back arrow icon')).toBeInTheDocument();
  });

  it('renders  switch icon', () => {
    const { getByTestId, getByAltText } = render(<Header />);
    expect(getByTestId('switchIcon')).toBeInTheDocument();
    expect(getByAltText('Switch icon')).toBeInTheDocument();
  });
  
  it('should have alt property', () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId('switchIcon')).toHaveAttribute('alt');
    expect(getByTestId('backArrowIcon')).toHaveAttribute('alt');
  });
});
