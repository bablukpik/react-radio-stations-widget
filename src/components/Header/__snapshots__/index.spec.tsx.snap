// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<DocumentFragment>
  <header
    class="widgetHeader"
  >
    <nav
      class="nav"
    >
      <button
        class="button"
        type="button"
      >
        <img
          alt="Back arrow icon"
          data-testid="backArrowIcon"
          src="back-arrow.png"
        />
      </button>
      <h1
        class="navHeading"
        data-testid="navHeading"
      >
        STATIONS
      </h1>
      <button
        class="button"
        type="button"
      >
        <img
          alt="Switch icon"
          data-testid="switchIcon"
          src="switch.png"
        />
      </button>
    </nav>
  </header>
</DocumentFragment>
`;
